buildpacks/nodejs-yarn/src/layers/deps.rs

```diff
@@ -90,14 +90,19 @@ impl Layer for DepsLayer {
 
 impl DepsLayer {
     fn install(&self, layer_path: &Path) -> Result<(), DepsLayerError> {
-        let mut args = vec!["install", "--frozen-lockfile"];
+        let mut args = vec!["install"];
         let cache_path = layer_path.join("cache");
         let cache_folder = cache_path.to_string_lossy().clone();
-        if !self.yarn_app_cache {
-            args.append(&mut vec!["--cache-folder", &cache_folder]);
-        }
         if self.yarn_major_version == "1" {
-            args.append(&mut vec!["--production", "false"]);
+            args.append(&mut vec![
+                "--cache-folder",
+                &cache_folder,
+                "--frozen-lockfile",
+                "--production",
+                "false",
```

Comment:
Yarn v1 expects either `--production` or `--production=true|false` according to [the docs](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-production-true-false)